### ISSUE TITLE
Better translations

### ISSUE THREAD

**Issue by phiphou** [electricitymaps/electricitymaps-contrib#7091]
Working on a french translation, I came to an issue.

The key "comes from" should give "vient de".

But the full sentence "... comes from solar" shoud give "vient DU solaire".

And if it comes from wind, it would give "vient DE L'élien".

And for geothermy, it would give "vient DE LA géothermie".

I finaly replaced the wording "comes from" by a translation for "is generated by".

It allows a better translation, "...comes from solar" becomes "...est générée par solaire".

But that's not perfect, the good translation should be "...est générée par DU solaire", or for wind "...est générée par DE L'éolien".

One solution would be to add a different translation keys for each energy sources, like this : 

```json
{
  "comesFrom_wind": "est générée par de l'",
  "comesFrom_solar": "est générée par du",
  "comesFrom_geothermy": "est générée par de la"
}
```

Any french speaker can give us his views on this point ?

It also may touch langages like german (der/die/das)...

**Comment by VIKTORVAV99** [comment]
@tonypls what is your take on this?

It means we would have to revise the logic on the tooltips a bit but it's doable.

**Comment by phiphou** [comment]
Please let me know when you dive in this ;)

**Comment by phiphou** [comment]
Forgot to mention the problem is the same with keys "electricityStoredUsing" and "emissionsStoredUsing".

"...is stored using wind" should give "...est stockée en utilisant [de l' / l'] éolien", and not "...est stockée en utilisant éolien".

"...is stored using solar" should give "...est stockée en utilisant [le / du] solaire", and not "...est stockée en utilisant solaire".

Use the "via" word may help, we would have "...est stockée via éolien". That's not perfect but probably better.

It would be usefull to have feedback on this from french speaker...

**Comment by VIKTORVAV99** [comment]
> Forgot to mention the problem is the same with keys "electricityStoredUsing" and "emissionsStoredUsing".
> 
> "...is stored using wind" should give "...est stockée en utilisant [de l' / l'] éolien", and not "...est stockée en utilisant éolien".
> 
> "...is stored using solar" should give "...est stockée en utilisant [le / du] solaire", and not "...est stockée en utilisant solaire".
> 
> Use the "via" word may help, we would have "...est stockée via éolien". That's not perfect but probably better.
> 
> It would be usefull to have feedback on this from french speakers...

I don't know French but how would this work with hydro and battery. Would it be the same for those? (those are the only storage modes we support at the moment).

**Comment by phiphou** [comment]
yes, It works the same, idealy we shoud apply the multiple keys solution for these two other keys.

But the semi-solution I proposed in my last PR (using the "via" word) works not so bad in these two cases. ("is stored using wind" becomes "est stockée via éolien").

It could be interessting to know if this "problem" appears in some other locales, I understand it may not be a good thing to change the logic for only one specific locale.

**Comment by phiphou** [comment]
Glad to know the problem is not for french only ;)

**Comment by madsnedergaard** [comment]
Thanks for the feedback here, we'll bring this forwards in future discussions around the translations as we intended to rework the whole setup in the future - but for now I'll close this issue to keep GitHub more focused around data issues and not feature requests :)
